src/ngResource/resource.js

```diff
@@ -663,11 +665,13 @@ angular.module('ngResource', ['ng']).
             }
 
             if (hasBody) httpConfig.data = data;
-            route.setUrlParams(httpConfig,
-              extend({}, extractParams(data, action.params || {}), params),
-              action.url);
-
-            var promise = $http(httpConfig).then(function(response) {
+            var promise = $q.all(extractParams(data, action.params || {})).then(function(ids) {
+              route.setUrlParams(httpConfig,
+                extend({}, ids, params),
```

Comment:
I don't think the first `{}` argument is needed any more.
(In fact, I don't think it was needed before anyway.)